dataops/dev_cicd_template.yml

```diff
@@ -2,13 +2,45 @@ variables:
   GITLAB_CICD_IMAGE_REPO: gcr.io/getindata-images-public/cicd-dataops
   GITLAB_CICD_IMAGE_TAG: gcp-0.1.1
   BLOB_CONFIG_PATH: blob_args.json
+  GIT_KEY_PATH: TBS
 
 dev:
   image: $GITLAB_CICD_IMAGE_REPO:$GITLAB_CICD_IMAGE_TAG
+  environment: DEVELOPMENT
   stage: execute-dev
   extends:
-    - .setup
+    - .dev_setup
   script:
-    - dp compile --docker-build
+    - dp compile --docker-build --env dev
     - dp deploy --blob-args $BLOB_CONFIG_PATH --datahub-ingest --docker-push
-    - dp publish --key-path $GIT_KEY_PATH
\ No newline at end of file
+  rules:
+    - if: '$CI_COMMIT_BRANCH != "master"'
+
+release-dev:
+  image: $GITLAB_CICD_IMAGE_REPO:$GITLAB_CICD_IMAGE_TAG
+  environment: DEVELOPMENT
+  stage: execute-release
```

Comment:
In case of this stages: 
  - execute-dev
  - execute-release

Gitlab will know to use the proper `execute-release` stage?

Because `release-dev` and `release-prod` have the same stage: `execute-release`

Reply:
I hope that will be 2 different tasks in one stage :) I will test is soon.